### PR TITLE
fix wrong native method overriding for iframe without src (close #279)

### LINKED ISSUE #279: Wrong native methods initialization for iframe without src
Test for reproduce
```javascript
asyncTest('wrong overrided native methods', function () {
   var iframe = document.createElement('iframe');

    iframe.id = 'test_unique_id_lkjlosjkf';
    iframe.addEventListener('load', function () {
        var iframeHammerhead = this.contentWindow['%hammerhead%'];
        var nativeAppendChild = iframeHammerhead.sandbox.nativeMethods.appendChild.toString();
        var overridedAppendChild = document.createElement('div').appendChild.toString();

        ok(nativeAppendChild !== overridedAppendChild);
        start();
    });
    document.body.appendChild(iframe);
});
```

Only in IE9, IE10
All element and document methods overrided wrongly.

### PR DESCRIPTION
**Pull request by miherlosev**
/cc @inikulin @churkin 
@VasilyStrelyaev check comments and test name

### CODE REVIEW

File: src/client/sandbox/native-methods.js [hunk 1]
```diff
@@ -138,6 +140,8 @@ class NativeMethods {
         // Event
         document.addEventListener    = this.documentAddEventListener;
         document.removeEventListener = this.documentRemoveEventListener;
+
+        this.isDocumentMethodsSaved  = false;
```
Comment (inikulin) [inline src/client/sandbox/native-methods.js]: I don't get it. Why we need this flag? And can we avoid adding it to the `nativeMethods`?

File: src/client/sandbox/node/index.js [hunk 2]
```diff
@@ -79,10 +80,13 @@ export default class NodeSandbox extends SandboxBase {
...
 
-        this.iframeSandbox.on(this.iframeSandbox.IFRAME_DOCUMENT_CREATED_EVENT, e =>
-                // NOTE: Override only document (In fact, we only need 'write' and 'writeln' methods).
-                this.doc.attach(e.iframe.contentWindow, e.iframe.contentDocument)
-        );
+        this.iframeSandbox.on(this.iframeSandbox.IFRAME_DOCUMENT_CREATED_EVENT, e => {
+            // NOTE: Save native document methods before overriding
+            nativeMethods.refreshDocumentMeths(e.iframe.contentDocument);
+
+            // NOTE: Override only document (In fact, we only need 'write' and 'writeln' methods).
```
Comment (VasilyStrelyaev) [inline src/client/sandbox/node/index.js]:
```
            // NOTE: Override only the document (in fact, we only need the 'write' and 'writeln' methods).
```

File: src/client/sandbox/native-methods.js (hunk 1 above)
Comment (miherlosev) [inline src/client/sandbox/native-methods.js]: Ok. I am changing it.

File: test/client/fixtures/sandbox/iframe-test.js [hunk 3]
```diff
@@ -200,3 +200,19 @@ asyncTest('iframe initialization must be synchronous (for iframes with an empty
     });
     document.body.appendChild(iframe);
 });
+
+asyncTest('wrong native methods initialization in iframe without src (GH-279)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandbox/iframe-test.js]: I guess the test checks that methods are initialized correctly - not wrongly
```
asyncTest('native methods are properly initialized in an iframe without src (GH-279)', function () {
```

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by miherlosev** [comment]
After corrector checking I've changed comment for https://github.com/miherlosev/testcafe-hammerhead/commit/92d70146dfbd17cc73a98947ae70f58d493f406d#diff-c2bc13f382998f8e54abf9b6f85c8101R83

**Comment by churkin** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm

**Pull request merged**